js/angular/app/scripts/modules/indicators/map-controller.js

```diff
@@ -173,17 +198,29 @@ angular.module('transitIndicators')
         OTIIndicatorManager.setConfig({calculation_job: calculation_job, city_name: ''}); // TODO: add city name
     });
 
-    // Bind events that will hide/show the jobs indicator legend when the jobs indicator is hidden/shown
+    // Bind events that will hide/show legends when the respective layer is hidden / shown
     leafletData.getMap().then(function(map) {
-        map.on('overlayadd', function(eventLayer) {
-            if (eventLayer.name === 'Jobs Indicator') {
-                $scope.jobsLegend = true;
+        var setLegendVisibility = function(layerName, legendState) {
+            switch(layerName) {
+                case $translate.instant('MAP.JOBS_INDICATOR_LAYER'):
```

Comment:
These translate calls are now being made in multiple places. Should save them to a variable.